front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/v2/ApplicationsControllerV2.groovy

```diff
@@ -0,0 +1,163 @@
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.validation.Errors
+import org.springframework.validation.ObjectError
+import org.springframework.web.bind.annotation.*
+
+import javax.servlet.http.HttpServletResponse
+
+@Slf4j
+@RestController
+@RequestMapping("/v2/applications")
+@Api(value = "application", description = "Application API")
+public class ApplicationsControllerV2 {
+  @Autowired
+  MessageSource messageSource
+
+  @Autowired
+  ApplicationDAO applicationDAO
+
+  @Autowired
+  ProjectDAO projectDAO
+
+  @Autowired
+  NotificationDAO notificationDAO
+
+  @Autowired
+  PipelineDAO pipelineDAO
+
+  @Autowired
+  PipelineStrategyDAO pipelineStrategyDAO
+
+  @Autowired
+  List<ApplicationValidator> applicationValidators
+
+  @Autowired(required = false)
+  List<ApplicationEventListener> applicationEventListeners = []
+
+  @Autowired
+  Registry registry
+
+  @PreAuthorize("@fiatPermissionEvaluator.storeWholePermission()")
+  @PostFilter("hasPermission(filterObject.name, 'APPLICATION', 'READ')")
+  @ApiOperation(value = "", notes = """Fetch all applications.
+
+    Supports filtering by one or more attributes:
+    - ?email=my@email.com
+    - ?email=my@email.com&name=flex""")
+  @RequestMapping(method = RequestMethod.GET)
+  Set<Application> applications(@RequestParam Map<String, String> params) {
+    if (params.isEmpty()) {
+      return applicationDAO.all()
+    }
+
+    return applicationDAO.search(params)
+  }
+
+  @PreAuthorize("hasPermission(#app.name, 'APPLICATION', 'CREATE')")
+  @ApiOperation(value = "", notes = "Create an application")
+  @RequestMapping(method = RequestMethod.POST, value = "")
+  Application create(@RequestBody final Application app) {
+    return getApplication().initialize(app).withName(app.getName()).save()
+  }
+
+  @PreAuthorize("hasPermission(#application, 'APPLICATION', 'WRITE')")
+  @ApiOperation(value = "", notes = "Delete an application")
+  @RequestMapping(method = RequestMethod.DELETE, value = "/{application}")
+  void delete(@PathVariable String application, HttpServletResponse response) {
+    getApplication().initialize(new Application().withName(application)).delete()
+    response.setStatus(HttpStatus.ACCEPTED.value())
+  }
+
+  @PreAuthorize("hasPermission(#app.name, 'APPLICATION', 'WRITE')")
+  @ApiOperation(value = "", notes = "Update an existing application")
+  @RequestMapping(method = RequestMethod.PUT, value = "/{name}")
+  Application update(@PathVariable String name, @RequestBody final Application app) {
+    if (name.trim() != app.getName()) {
+      log.error("non matching application name {}", name)
+      throw new InvalidApplicationRequestException("provided application name does not match")
+    }
+
+    def application = getApplication()
+    Application existingApplication = application.findByName(app.getName())
+    application.initialize(existingApplication).withName(app.getName()).update(app)
```

Comment:
This seems a bit odd to me that someone could update the name of the app config accidentally. Say someone `curl -X PUT /v2/applications/abc -d '{"name":"abcd"}' Am I reading this correctly?

@ajordens